src/Servers/Kestrel/Core/test/SniOptionsSelectorTests.cs

```diff
@@ -761,7 +760,7 @@ public X509Certificate2 LoadCertificate(CertificateConfig certInfo, string endpo
                     return null;
                 }
 
-                var cert = new X509Certificate2();
+                var cert = TestResources.GetTestCertificate();
```

Comment:
@halter73 introduced IsTestMock because he wasn't using a real cert here. Something about not wanting to create a large number of real certs.

Reply:
Is that a real problem?

Reply:
When @halter73 is back, if the test haven't exploded by then can give me some feedback.

Reply:
I'd recommend storing the test cert in a static like we do for some [other tests](https://github.com/dotnet/aspnetcore/blob/b2c7d514c928d572dcab58f4670578c7e40ddfbe/src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/TlsTests.cs#L29) and removing `IsTestMock` since there's no longer an issue with this returning invalid certs. I guess it's no big deal if it isn't causing any slowness or reliability issues though.

Reply:
The second PR does it. I'll do the work in there.